crates/oxc_semantic/src/builder.rs

```diff
@@ -580,6 +581,12 @@ impl<'a> SemanticBuilder<'a> {
                         }
                     }
                 }
+
+                if let Some(decl) = &decl.declaration {
+                    decl.bound_names(&mut |ident| {
+                        self.add_export_flag_to_identifier(ident.name.as_str());
+                    });
+                }
```

Comment:
Do you prefer to handle VariableDeclarations here only?

Reply:
Or do you prefer to handle all exports here (leave program), removing other code logic for handling exports?

Reply:
I prefer to handle all export declarations here because the previous way we need to avoid adding `SymbolFlags::Export` to the wrong symbol.

Reply:
@Dunqing I attempted to relocate some code but encountered strange problems. 
Directly moving the code from `add_export_flag_to_export_identifiers` (the `AstKind::Program` case of `leave_kind`) to `leave_kind`  is not equivalent.

I successfully moved `AstKind::ExportNamedDeclaration`, but when I tried to move `AstKind::ExportDefaultDeclaration`, the test cases failed.

Reply:
￼It has been a long time since I last modified this part code, and I need to get reacquainted with it. BTW, our team wants to fix this issue as soon as possible, I will make edits directly in this PR to avoid communication costs. I hope you don’t mind!